### PR TITLE
Shell + Message Router integration

### PR DESCRIPTION
**Pull request by BalassaMarton**
Message Router can now be created without any configuration in JS, just call `createMessageRouter()`.
Fixed async start and stop logic in Shell

### CODE REVIEW

File: Tryouts/Messaging-JS/.vscode/templates/js.lict [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Morgan Stanley makes this available to you under the Apache License,
```
Comment (psmulovics) [inline Tryouts/Messaging-JS/.vscode/templates/js.lict:1]: Should we add lict files for the .NET part too?

Comment (BalassaMarton) [reply Tryouts/Messaging-JS/.vscode/templates/js.lict:1]: We can, but I've been adding the license headers with ReSharper, and they are not compatible. I'm not sure if the vscode extension I used here is something widely used or just a random first hit.

File: Tryouts/Prototypes/Shell/App.xaml.cs [hunk 2]
```diff
@@ -12,77 +12,160 @@
...
+        CreateWindow<WebWindow>(options).Show();
+    }
+
+    private void InjectMessageRouterConfig()
+    {
+        var server = Host.Services.GetRequiredService<IMessageRouterWebSocketServer>();
+        _logger.LogInformation($"Message Router server listening at {server.WebSocketUrl}");
+
+        WebWindow.AddPreloadScript(
+            $$"""
```
[the remark this answers is not in the document]
Comment (BalassaMarton) [reply Tryouts/Prototypes/Shell/App.xaml.cs:157]: Hm, `LangVersion` is set to `latest` so maybe the pipeline uses old tooling?

[the remark this answers is not in the document]
Comment (BalassaMarton) [reply Tryouts/Prototypes/Shell/App.xaml.cs:157]: OK so C# 11 is included in .NET 7 SDK, so we can use that SDK in the pipeline, or I can revert this to using the old syntax.

[the remark this answers is not in the document]
Comment (BalassaMarton) [reply Tryouts/Prototypes/Shell/App.xaml.cs:157]: Setting `LangVersion` to `preview` solved the problem.

**Comment by psmulovics** [comment]
"Chrome 110.0.0.0 (Windows 10) DatagridComponent it should create app FAILED
	Error: Unable to create the MessageRouter client, configuration is missing." from the build

**Comment by BalassaMarton** [comment]
> "Chrome 110.0.0.0 (Windows 10) DatagridComponent it should create app FAILED Error: Unable to create the MessageRouter client, configuration is missing." from the build

Yup, forgot to run those tests. I'll fix the tests, `MessageRouter` should be injected and mocked instead of creating it in each component.

File: Tryouts/Plugins/ApplicationPlugins/datagrid/package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "@angular/platform-browser": "^15.1.0",
     "@angular/platform-browser-dynamic": "^15.1.0",
     "@angular/router": "^15.1.0",
-    "material-icons": "1.13.1",
```
Comment (kruplm) [inline Tryouts/Plugins/ApplicationPlugins/datagrid/package.json:23]: We need material icons

Comment (BalassaMarton) [reply Tryouts/Plugins/ApplicationPlugins/datagrid/package.json:23]: Fixed.

File: Tryouts/Messaging-JS/package.json [hunk 4]
```diff
@@ -1,25 +1,36 @@
...
+    "type": "module",
+    "files": [
+        "dist/esm/**/*"
+    ],
+    "main": "dist/esm/index.js",
+    "module": "dist/esm/index.js",
+    "types": "dist/esm/index.d.ts",
     "scripts": {
-        "build": "yarn run build:clean && yarn run build:esm && yarn run build:cjs",
+        "build": "yarn && yarn run build:clean && yarn run build:rollup",
```
Comment (kruplm) [inline Tryouts/Messaging-JS/package.json:13]: if you want to use yarn then it should be

- a) added as a devDependency
- b) invoked with npx

Unless you do that everyone in the dev team has to globally install yarn first

**Review by kruplm — CHANGES_REQUESTED** [review]

File: Tryouts/Messaging-JS/package.json (hunk 4 above)
Comment (BalassaMarton) [reply Tryouts/Messaging-JS/package.json:13]: Understood, removing the dependency on yarn

Comment (kruplm) [reply Tryouts/Messaging-JS/package.json:13]: ok. So delete yar.lock from Messaging-JS
and the build script should look like:  "build": "npm run build:clean && npm run build:rollup",

**Review by kruplm — APPROVED** [review]

**Pull request merged**